ci-operator/step-registry/openstack/conf/clouds/openstack-conf-clouds-commands.sh

```diff
@@ -7,14 +7,27 @@ set -o pipefail
 CLUSTER_TYPE="${CLUSTER_TYPE_OVERRIDE:-$CLUSTER_TYPE}"
 export OS_CLIENT_CONFIG_FILE="${SHARED_DIR}/clouds.yaml"
 
+is_openshift_version_gte() {
+	declare release_image ocp_version
+	release_image="$(openshift-install version | sed -n 's/^release image\s\+\(.*\)$/\1/p' | tr -d '\n')"
+	ocp_version="$(oc adm release info "$release_image" -o json | jq -r '.metadata.version' | tr -d '\n')"
+	echo "Detected OCP version: ${ocp_version}"
+	printf '%s\n%s' "$1" "$ocp_version" | sort -C -V
+}
+
 cp "/var/run/cluster-secrets/${CLUSTER_TYPE}/clouds.yaml" "$OS_CLIENT_CONFIG_FILE"
 
 if [ -f "/var/run/cluster-secrets/${CLUSTER_TYPE}/osp-ca.crt" ]; then
 	cp "/var/run/cluster-secrets/${CLUSTER_TYPE}/osp-ca.crt" "${SHARED_DIR}/osp-ca.crt"
 	sed -i "s|cacert: .*|cacert: ${SHARED_DIR}/osp-ca.crt|" "$OS_CLIENT_CONFIG_FILE"
 fi
 
-if [[ -n $APPLICATION_CREDENTIALS ]]; then
+if ! is_openshift_version_gte "4.12"; then
```

Comment:
Could you add a comment explaining the reasons we have this check? Could be as simple as saying that application credentials is supported starting from 4.12.